src/firefly/js/visualize/rawData/RawImageTilesGPU.js

```diff
@@ -1,187 +1,95 @@
-            const pixel= pixelAry[height-this.thread.y-1][this.thread.x];
-            let pixelIdx= pixel*3;
-            this.color( colorModel[pixelIdx], colorModel[pixelIdx+1], colorModel[pixelIdx+2]);
-            `
-    );
-
-    const standardFuncWithContrast= Function('pixelAry', 'colorModel','height', 'contrast', 'offsetShift', `
-            const pixel= pixelAry[height-this.thread.y-1][this.thread.x];
-            let pixelIdx= pixel*3;
-            const newPixel = Math.floor( offsetShift+(pixel*contrast));
-            pixelIdx= (newPixel > 254) ? 762 : (newPixel<0) ? 0 : newPixel*3;
-            this.color( colorModel[pixelIdx], colorModel[pixelIdx+1], colorModel[pixelIdx+2]);
-            `
-    );
-
-    const standRawDataTileSimpleGPU = gpu.createKernel(standardFuncSimple,{graphical:true, dynamicOutput:true, dynamicArguments:true, tactic: 'speed'});
-    const standRawDataTileWithContrastGPU = gpu.createKernel(standardFuncWithContrast,{graphical:true, dynamicOutput:true, dynamicArguments:true, tactic: 'speed'});
-
-    const threeCFunc = Function(
-        'redAry', 'greenAry', 'blueAry', 'use', 'contrast', 'offsetShift', 'width','height', `
-        const idx= (height - this.thread.y-1) * width + this.thread.x;
-        let r= use[0]===1 ? redAry[idx] : 0;
-        let g= use[1]===1 ? greenAry[idx] : 0;
-        let b= use[2]===1 ? blueAry[idx] : 0;
-        if (r!==0 && offsetShift[0]!==0) {
-            r= Math.floor( offsetShift[0]+(r*contrast[0]));
-            r= (r > 254) ? 254 : (r<0) ? 0 : r;
-        }
-        if (g!==0 && offsetShift[1]!==0) {
-            g= Math.floor( offsetShift[1]+(g*contrast[1]));
-            g= (g > 254) ? 254 : (g<0) ? 0 : g;
-        }
-        if (b!==0&& offsetShift[2]!==0) {
-            b= Math.floor( offsetShift[2]+(b*contrast[2]));
-            b= (b > 254) ? 254 : (b<0) ? 0 : b;
-        }
-        this.color( r/255, g/255, b/255, 1);`
-    );
-
-    const threeCRawDataTileGPU = gpu.createKernel(threeCFunc,{graphical:true, dynamicOutput:true, dynamicArguments:true, tactic: 'speed'});
-
-    async function createTransitionalTileWithGPU(inData, colorModel, isThreeColor, mask, maskColor, bias=.5, contrast=1, bandUse) {
-        const canvas= createTileWithGPU(inData,colorModel,mask, maskColor, bias,contrast, bandUse);
-
-        const g= getGlobalObj();
-        if (inWorker() && g.createImageBitmap) { // if in a worker then turn canvas into ImageBitmap to return to main
-            const workerTmpTile= await g.createImageBitmap(canvas);
-            return { ...inData, workerTmpTile };
-        }
-        else {
-            return { ...inData, workerTmpTile:canvas };
-        }
+import {gpujs_createRawDataTile3CRGB, gpujs_createRawDataTileImageRGB} from './GpuJsJobs';
+import {process3CTileViaWebGPU, processMaskTileViaWebGPU, processTileViaWebGPU} from './WebGpuJobs';
+
+const getTilingFuncs= once(() => {
+    if (BrowserInfo.supportsWebGpu() ) {
```

Comment:
this fallback is great! But what happens if a browser meets the version requirement (like chrome) but still gpu adaptor couldn't be found/initialized? See my related comment in BrowserInfo.js which might be the solution?!